### PR TITLE
feat: add getter to return all source chain config

### PR DESCRIPTION
**Pull request by 0xsuryansh**
## Motivation
The current offchain initialization process involves calling getSourceChainConfig(sourceChainSelector), which retrieves configurations for a specific source chain selector. However, there's no mechanism to fetch a complete list of supported source chains during the initialization phase. This limitation complicates the process, as the full list of source chain selectors is not readily available.

## Solution
To address this, the proposed solution introduces an enumerable set of all source chain selectors. By implementing a getAllSourceChainConfig() function, we can easily retrieve the complete list of source chains on-chain. This eliminates the need for additional rounds of OCR gossip to share known source chains, simplifying initialization and improving efficiency.

### CODE REVIEW

File: contracts/src/v0.8/ccip/offRamp/OffRamp.sol [hunk 1]
```diff
@@ -919,6 +924,16 @@ contract OffRamp is ITypeAndVersion, MultiOCR3Base {
     return s_sourceChainConfigs[sourceChainSelector];
   }
 
+  /// @notice Returns all source chain configs
+  /// @return sourceChainConfigs The source chain configs corresponding to all the supported chain selectors
+  function getAllSourceChainConfig() external view returns (SourceChainConfig[] memory) {
```
Comment (RensR) [inline contracts/src/v0.8/ccip/offRamp/OffRamp.sol]: This doesn't return the chain selectors, which I would suspect are required?

File: contracts/src/v0.8/ccip/offRamp/OffRamp.sol [hunk 2]
```diff
@@ -931,6 +946,7 @@ contract OffRamp is ITypeAndVersion, MultiOCR3Base {
     for (uint256 i = 0; i < sourceChainConfigUpdates.length; ++i) {
       SourceChainConfigArgs memory sourceConfigUpdate = sourceChainConfigUpdates[i];
       uint64 sourceChainSelector = sourceConfigUpdate.sourceChainSelector;
+      s_sourceChainSelectors.add(sourceChainSelector);
```
Comment (RensR) [inline contracts/src/v0.8/ccip/offRamp/OffRamp.sol]: Lets do this after the sanity checks

File: contracts/src/v0.8/ccip/offRamp/OffRamp.sol (hunk 1 above)
Comment (0xsuryansh) [reply contracts/src/v0.8/ccip/offRamp/OffRamp.sol]: https://github.com/smartcontractkit/ccip/pull/1477/commits/094bdc51758e36eb3705355c618d90a49daf07a5

File: contracts/src/v0.8/ccip/offRamp/OffRamp.sol (hunk 2 above)
Comment (0xsuryansh) [reply contracts/src/v0.8/ccip/offRamp/OffRamp.sol]: https://github.com/smartcontractkit/ccip/pull/1477/commits/1d42a4ab7a832f4fdaab7b5f43f266f76e511cc3

File: contracts/src/v0.8/ccip/offRamp/OffRamp.sol [hunk 3]
```diff
@@ -919,6 +924,18 @@ contract OffRamp is ITypeAndVersion, MultiOCR3Base {
     return s_sourceChainConfigs[sourceChainSelector];
   }
 
+  /// @notice Returns all source chain configs
+  /// @return sourceChainConfigs The source chain configs corresponding to all the supported chain selectors
+  function getAllSourceChainConfig() external view returns (uint64[] memory, SourceChainConfig[] memory) {
```
Comment (makramkd) [inline contracts/src/v0.8/ccip/offRamp/OffRamp.sol]: naming nit: `getAllSourceChainConfigs` (with an `s` at the end)? Whats the preference for this @RensR

Comment (RensR) [reply contracts/src/v0.8/ccip/offRamp/OffRamp.sol]: Its a list so yeah good call

File: contracts/src/v0.8/ccip/offRamp/OffRamp.sol [hunk 4]
```diff
@@ -962,6 +979,8 @@ contract OffRamp is ITypeAndVersion, MultiOCR3Base {
       currentConfig.isEnabled = sourceConfigUpdate.isEnabled;
       currentConfig.router = sourceConfigUpdate.router;
 
+      s_sourceChainSelectors.add(sourceChainSelector);
```
Comment (RensR) [inline contracts/src/v0.8/ccip/offRamp/OffRamp.sol:983]: nit: this returns a bool for "inserted" or "already exists". maybe add a comment explaining we don't care about the return value as adding it twice should not do anything

File: contracts/src/v0.8/ccip/offRamp/OffRamp.sol (hunk 3 above)
Comment (0xsuryansh) [reply contracts/src/v0.8/ccip/offRamp/OffRamp.sol]: https://github.com/smartcontractkit/ccip/pull/1477/commits/e053e9eebd9a87666842548697768bd289e23d25#diff-865ecf8626037bea80482f856f6aff9bdd5a948982897dab6e2b4ff335af032aR929

File: contracts/src/v0.8/ccip/offRamp/OffRamp.sol (hunk 4 above)
Comment (0xsuryansh) [reply contracts/src/v0.8/ccip/offRamp/OffRamp.sol:983]: https://github.com/smartcontractkit/ccip/pull/1477/commits/e053e9eebd9a87666842548697768bd289e23d25#diff-865ecf8626037bea80482f856f6aff9bdd5a948982897dab6e2b4ff335af032aR982

**Review by makramkd — APPROVED** [review]

File: contracts/src/v0.8/ccip/offRamp/OffRamp.sol (hunk 4 above)
Comment (0xsuryansh) [reply contracts/src/v0.8/ccip/offRamp/OffRamp.sol:983]: https://github.com/smartcontractkit/ccip/pull/1477/commits/e053e9eebd9a87666842548697768bd289e23d25#diff-865ecf8626037bea80482f856f6aff9bdd5a948982897dab6e2b4ff335af032aR982

**Review by RensR — APPROVED** [review]

**Pull request merged**